config/realm.go

```diff
@@ -1,14 +1,40 @@
 package config
 
-import "github.com/frain-dev/convoy/auth"
+import (
+	"encoding/json"
+
+	"github.com/frain-dev/convoy/auth"
+)
+
+type BasicAuthConfig []BasicAuth
 
 type BasicAuth struct {
 	Username string    `json:"username"`
 	Password string    `json:"password"`
 	Role     auth.Role `json:"role"`
 }
 
+// Decode loads in config from an env var named `CONVOY_BASIC_AUTH_CONFIG`
+func (b *BasicAuthConfig) Decode(value string) error {
+	config := BasicAuthConfig{}
+	err := json.Unmarshal([]byte(value), &config)
+
+	*b = config
+	return err
+}
+
+type APIKeyAuthConfig []APIKeyAuth
+
 type APIKeyAuth struct {
 	APIKey string    `json:"api_key"`
 	Role   auth.Role `json:"role"`
 }
```

Comment:
This takes away the ability to set multiple right?

Reply:
Nah, we can still set mutiple like this 
```sh
export CONVOY_BASIC_AUTH_CONFIG="[{\"username\": \"some-admin\",\"password\": \"some-password\",\"role\": {\"type\": \"super_user\",\"groups\": []}}]"
export CONVOY_API_KEY_CONFIG="[{\"api_key\":\"ABC1234\",\"role\":{\"type\":\"admin\",\"groups\":[\"group-uid-1\",\"group-uid-2\"],\"apps\":[\"apps-uid-1\",\"apps-uid-2\"]}}]"
```